apps/intellij/src/main/kotlin/dev/nx/console/toolWindow/NxProjectsTreeStructure.kt

```diff
@@ -0,0 +1,327 @@
+package dev.nx.console.toolWindow
+
+import com.intellij.execution.Executor
+import com.intellij.execution.RunManager
+import com.intellij.execution.RunnerAndConfigurationSettings
+import com.intellij.execution.executors.DefaultRunExecutor
+import com.intellij.execution.impl.RunDialog
+import com.intellij.icons.AllIcons
+import com.intellij.lang.javascript.JavaScriptBundle
+import com.intellij.openapi.Disposable
+import com.intellij.openapi.actionSystem.*
+import com.intellij.openapi.components.service
+import com.intellij.openapi.project.DumbAware
+import com.intellij.ui.ClickListener
+import com.intellij.ui.PopupHandler
+import com.intellij.ui.tree.AsyncTreeModel
+import com.intellij.ui.tree.StructureTreeModel
+import com.intellij.ui.treeStructure.CachingSimpleNode
+import com.intellij.ui.treeStructure.SimpleNode
+import com.intellij.ui.treeStructure.SimpleTreeStructure
+import com.intellij.util.ui.tree.TreeUtil
+import dev.nx.console.NxIcons
+import dev.nx.console.models.NxProject
+import dev.nx.console.models.NxWorkspace
+import dev.nx.console.run.NxCommandConfiguration
+import dev.nx.console.run.NxCommandConfigurationType
+import dev.nx.console.run.NxRunSettings
+import java.awt.event.MouseEvent
+
+class NxProjectsTreeStructure(
+    val nxExecutor: NxExecutor,
+    val tree: NxProjectsTree,
+    val parentDisposable: Disposable,
+    private var nxWorkspace: NxWorkspace?
+) : SimpleTreeStructure() {
+
+    private val treeModel = StructureTreeModel(this, parentDisposable)
+    private var root: NxSimpleNode.Root = NxSimpleNode.Root(nxWorkspace)
+
+    init {
+        tree.model = AsyncTreeModel(treeModel, parentDisposable)
+        TreeUtil.installActions(tree)
+        installPopupActions()
+    }
+
+    override fun getRootElement(): Any = root
+
+    fun updateNxProjects(nxWorkspace: NxWorkspace?) {
+        this.nxWorkspace = nxWorkspace
+        root = NxSimpleNode.Root(nxWorkspace)
+        treeModel.invalidate()
+    }
+
+    sealed class NxSimpleNode(parent: SimpleNode?) : CachingSimpleNode(parent) {
+        class Root(val nxWorkspace: NxWorkspace?) : NxSimpleNode(null) {
+
+            init {
+                icon = NxIcons.Action
+            }
+
+            override fun buildChildren(): Array<SimpleNode> {
+                if (nxWorkspace == null) {
+                    return emptyArray()
+                }
+                val tasks: Array<SimpleNode> = arrayOf(Tasks(nxWorkspace, this))
```

Comment:
The Tree itself looks good. But we are moving more and more away from having these forced `apps` and `libs` directories. So it should be structured differently. The top-level nodes should be `Targets` and `Projects`.

The targets are already displayed well. For the projects, we have two options for displaying them: by folder structure or just having a list. In VSCode we call this "list view" and "tree view". 
You don't have to implement it like this right away, just pick one (list view is easier) and build it. Examples below 👇 

### Example of the tree view:
![image](https://user-images.githubusercontent.com/34165455/225066417-d29a25c1-173d-4b4b-a3ce-283f5a02d0c8.png)
you can see the structure is the same as the folders

### Example of the list view:
![image](https://user-images.githubusercontent.com/34165455/225067170-39952f09-35ee-49d6-b1f3-ae87c4813381.png)
you can see that everything is just one long list.

Reply:
the apps and lib are not the directories but rather the types of nx projects
we can have some settings that says for Projects group by project type, by path ...